network-store-server/src/main/java/com/powsybl/network/store/server/NetworkStoreRepository.java

```diff
@@ -407,34 +407,74 @@ public void deleteNetwork(UUID uuid, int variantNum) {
         }
     }
 
-    @SuppressWarnings("javasecurity:S5145")
-    public void cloneNetwork(UUID uuid, int sourceVariantNum, int targetVariantNum, String targetVariantId) {
+    public void cloneNetwork(UUID targetNetworkUuid, UUID sourceNetworkUuid, List<String> targetVariantIds) {
+        List<VariantInfos> networkVariantsInfo = getVariantsInfos(sourceNetworkUuid).stream()
+                .filter(v -> targetVariantIds.contains(v.getId()))
+                .sorted(Comparator.comparing(VariantInfos::getNum))
+                .collect(Collectors.toList());
+
+        targetVariantIds.stream()
+                .filter(element -> !networkVariantsInfo.stream()
+                        .map(VariantInfos::getId)
+                        .collect(Collectors.toList()).contains(element))
+                .forEach(variantNotFound ->
+                        LOGGER.warn("The network {} has no variant ID named : {}, thus it won't be cloned", sourceNetworkUuid, variantNotFound));
+
+        networkVariantsInfo.forEach(variantInfos -> {
+            Resource<NetworkAttributes> sourceNetworkAttribute = getNetwork(sourceNetworkUuid, variantInfos.getNum()).orElseThrow(() -> new PowsyblException("Cannot retrieve source network attributes uuid : " + sourceNetworkUuid + ", variantId : " + variantInfos.getId()));
+            sourceNetworkAttribute.getAttributes().setUuid(targetNetworkUuid);
+            sourceNetworkAttribute.setVariantNum(networkVariantsInfo.indexOf(variantInfos));
```

Comment:
You have to use the same also as the one on client side to find a free variantNum. Look at @jonenst PR #255, has moved the also to a utility class.